hudi-client/hudi-java-client/src/test/java/org/apache/hudi/hadoop/TestHoodieFileGroupReaderOnHive.java

```diff
@@ -0,0 +1,329 @@
+    try {
+      //prepare fg reader records to be compared to the baseline reader
+      HoodieReaderContext<ArrayWritable> readerContext = getHoodieReaderContext(tablePath, schema, storageConf);
+      Map<String, ArrayWritable> recordMap = new HashMap<>();
+      for (ArrayWritable record : actualRecordList) {
+        recordMap.put(readerContext.getRecordKey(record, schema), record);
+      }
+
+
+      JobConf jobConf = new JobConf(baseJobConf);
+      jobConf.set(HoodieReaderConfig.FILE_GROUP_READER_ENABLED.key(), "false");
+
+      TableDesc tblDesc = Utilities.defaultTd;
+      // Set the input format
+      tblDesc.setInputFileFormatClass(HoodieParquetRealtimeInputFormat.class);
+      LinkedHashMap<Path, PartitionDesc> pt = new LinkedHashMap<>();
+      LinkedHashMap<Path, ArrayList<String>> talias = new LinkedHashMap<>();
+
+      PartitionDesc partDesc = new PartitionDesc(tblDesc, null);
+
+      pt.put(new Path(tablePath), partDesc);
+
+      ArrayList<String> arrayList = new ArrayList<>();
+      arrayList.add(tablePath);
+      talias.put(new Path(tablePath), arrayList);
+
+      MapredWork mrwork = new MapredWork();
+      mrwork.getMapWork().setPathToPartitionInfo(pt);
+      mrwork.getMapWork().setPathToAliases(talias);
+
+      Path mapWorkPath = new Path(tablePath);
+      Utilities.setMapRedWork(jobConf, mrwork, mapWorkPath);
+
+      // Add three partition path to InputPaths
+      Path[] partitionDirArray = new Path[HoodieTestDataGenerator.DEFAULT_PARTITION_PATHS.length];
+      Arrays.stream(HoodieTestDataGenerator.DEFAULT_PARTITION_PATHS).map(s -> new Path(tablePath, s)).collect(Collectors.toList()).toArray(partitionDirArray);
+      FileInputFormat.setInputPaths(jobConf, partitionDirArray);
+      jobConf.set(HAS_MAP_WORK, "true");
+      // The following config tells Hive to choose ExecMapper to read the MAP_WORK
+      jobConf.set(MAPRED_MAPPER_CLASS, ExecMapper.class.getName());
+      // setting the split size to be 3 to create one split for 3 file groups
+      jobConf.set(org.apache.hadoop.mapreduce.lib.input.FileInputFormat.SPLIT_MAXSIZE, "128000000");
+      setupJobconf(jobConf);
+
+      HoodieCombineHiveInputFormat combineHiveInputFormat = new HoodieCombineHiveInputFormat();
+      InputSplit[] splits = combineHiveInputFormat.getSplits(jobConf, 1);
+
+      assertEquals(1, splits.length);
+      RecordReader<NullWritable, ArrayWritable> reader = combineHiveInputFormat.getRecordReader(splits[0], jobConf, Reporter.NULL);
+      // use reader to read log file.
+      NullWritable key = reader.createKey();
+      ArrayWritable value = reader.createValue();
+      while (reader.next(key, value)) {
+        if (readerContext.getValue(value, schema, HoodieRecord.FILENAME_METADATA_FIELD).toString().contains(fileGroupId)) {
+          //only evaluate records from the specified filegroup. Maybe there is a way to get
+          //hive to do this?
```

Comment:
Could you filter the records based on `FILENAME_METADATA_FIELD` to get the records for a particular file group?

Reply:
That's what I'm suggesting. But IDK how to do that. None of the other tests do that and I can't figure out. Since we have a solution that works fine 
```
readerContext.getValue(value, schema, HoodieRecord.FILENAME_METADATA_FIELD).toString().contains(fileGroupId)
```
I don't think it's worth it to spend more time on this

Reply:
Makes sense.